src/util/bit_util.rs

```diff
@@ -250,13 +250,13 @@ impl BitWriter {
   }
 
   /// Write the `num_bits` LSB of value `v` to the internal buffer of this writer.
-  /// The `num_bits` must not be greater than 32. This is bit packed.
+  /// The `num_bits` must not be greater than 64. This is bit packed.
   ///
   /// Return false if there's not enough room left. True otherwise.
   #[inline]
   pub fn put_value(&mut self, v: u64, num_bits: usize) -> bool {
-    assert!(num_bits <= 32);
-    assert_eq!(v >> num_bits, 0);
+    assert!(num_bits <= 64);
+    assert_eq!(v.checked_shr(num_bits as u32).unwrap_or(0), 0); // covers case v >> 64
```

Comment:
why is `num_bits = 64` not covered by `v >> num_bits`?

Reply:
I was puzzled by this as well, potentially because I do not really understand how languages implement right shift. It looks like right shift returns the same result when shift >= width of type - it is okay to shift 63 and 1 for example, but is not okay to shift 64; and it is behaviour for most of the languages out there (found related posts for C++/C, and Java).

Here is Java explanation on this (http://docs.oracle.com/javase/specs/jls/se7/html/jls-15.html#jls-15.19):
> If the promoted type of the left-hand operand is int, only the five lowest-order bits of the right-hand operand are used as the shift distance. It is as if the right-hand operand were subjected to a bitwise logical AND operator & (§15.22.1) with the mask value 0x1f (0b11111). The shift distance actually used is therefore always in the range 0 to 31, inclusive.

> If the promoted type of the left-hand operand is long, then only the six lowest-order bits of the right-hand operand are used as the shift distance. It is as if the right-hand operand were subjected to a bitwise logical AND operator & (§15.22.1) with the mask value 0x3f (0b111111). The shift distance actually used is therefore always in the range 0 to 63, inclusive. 

I found that Rust has a function `checked_shr` that returns None when such shift is performed, therefore used it to cover that case.